app/src/main/java/com/github/libretube/ui/controllers/FullscreenGestureAnimationController.kt

```diff
@@ -0,0 +1,96 @@
+package com.github.libretube.ui.controllers
+
+import android.view.MotionEvent
+import android.view.animation.AccelerateDecelerateInterpolator
+import com.github.libretube.extensions.dpToPx
+import com.github.libretube.ui.views.CustomExoPlayerView
+
+/**
+ * A class that handles the scale animation in [CustomExoPlayerView] when entering/exiting
+ * fullscreen using the swipe gesture.
+ */
+class FullscreenGestureAnimationController(
+    private val playerView: CustomExoPlayerView,
+    private val onSwipeUpCompleted: () -> Unit,
+    private val onSwipeDownCompleted: () -> Unit,
+) {
+    private var isSwipeInProgress = false
+    private var shouldHandleSwipe = false
+    /**
+     * - `value > 0` = Swiping up
+     * - `value < 0` = Swiping down
+     */
+    private var swipeDirection = 0
+    private var swipeStartY = 0f
+
+    fun onSwipe(distanceY: Float, e2: MotionEvent) {
```

Comment:
```suggestion
    fun onSwipe(distanceY: Float, positionY: Float) {
```
Since we only use `y` from `e2`, we should use the information hiding principle and only pass the position `e2.y` here instead of the full event